packages/panzer/disc-fe/src/lof/Panzer_EpetraLinearObjFactory_impl.hpp

```diff
@@ -633,32 +695,69 @@ const Teuchos::RCP<Epetra_Map> EpetraLinearObjFactory<Traits,LocalOrdinalT>::bui
-
-   // get the global indices
-   gidProvider_->getOwnedAndGhostedIndices(indices);
-
-   return Teuchos::rcp(new Epetra_Map(-1,indices.size(),&indices[0],0,*comm_));
-}
-
-// build the ghosted map
-template <typename Traits,typename LocalOrdinalT>
-const Teuchos::RCP<Epetra_Map> EpetraLinearObjFactory<Traits,LocalOrdinalT>::buildGhostedColMap() const
-{
-   if(!hasColProvider_)  
-     return buildGhostedMap();
-
-   std::vector<int> indices;
-
-   // get the global indices
-   colGidProvider_->getOwnedAndGhostedIndices(indices);
-
-   return Teuchos::rcp(new Epetra_Map(-1,indices.size(),&indices[0],0,*comm_));
-}
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedMap()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map> EpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildGhostedMap() const
+{
+  std::vector<int> indices;
+  gidProvider_->getOwnedAndGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *comm_));
+} // end of buildGhostedMap()
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedMap2()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map> EpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildGhostedMap2() const
+{
+  std::vector<int> indices;
+  gidProvider_->getGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *comm_));
+} // end of buildGhostedMap2()
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedColMap()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map> EpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildGhostedColMap() const
+{
+  if (!hasColProvider_)
+    return buildGhostedMap();
+  std::vector<int> indices;
+  colGidProvider_->getOwnedAndGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *comm_));
+} // end of buildGhostedColMap()
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedColMap2()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map> EpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildGhostedColMap2() const
+{
+  if (!hasColProvider_)
+    return buildGhostedMap2();
+  std::vector<int> indices;
+  colGidProvider_->getGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *comm_));
+} // end of buildGhostedColMap2()
```

Comment:
Same as original, but either calls `buildGhostedMap2`, or calls `getGhostedIndices` instead of `getOwnedAndGhostedIndices`.